### PR TITLE
feat: Implement AppProject watch to retrigger reconciliation

### PR DESCRIPTION
**Pull request by leoluz**
This PR implements a watch in the AppProject to invoke the AccessRequest reconciliation on changes in the managed roles.

### CODE REVIEW

File: internal/controller/accessrequest_controller.go [hunk 1]
```diff
@@ -55,6 +55,7 @@ const (
 	// managed by this controller
 	AccessRequestFinalizerName = "accessrequest.ephemeral-access.argoproj-labs.io/finalizer"
 	roleTemplateField          = ".spec.roleTemplateName"
+	projectField               = ".status.TargetProject"
```
Comment (crenshaw-dev) [inline internal/controller/accessrequest_controller.go:58]: Not a huge deal, but the capitalized first letter in `TargetProject` is a bit odd.

File: internal/controller/config/config.go [hunk 2]
```diff
@@ -26,21 +26,21 @@ type LogConfigurer interface {
 
 // MetricsConfigurer defines the accessor methods for metrics configurations.
 type MetricsConfigurer interface {
-	MetricsAddress() string
+	MetricsAddress() int
```
Comment (crenshaw-dev) [inline internal/controller/config/config.go:29]: Should the field be renamed to `Port`?

**Review by crenshaw-dev — CHANGES_REQUESTED** [review]
Just a couple small notes.

File: internal/controller/accessrequest_controller.go (hunk 1 above)
Comment (leoluz) [reply internal/controller/accessrequest_controller.go:58]: Not odd. It was wrong! :P 
Fixed :)

File: internal/controller/config/config.go (hunk 2 above)
Comment (leoluz) [reply internal/controller/config/config.go:29]: Oh.. This was actually a bad refactoring as users wouldn't be able to provide the address anymore. 
Will revert.

**Review by crenshaw-dev — APPROVED** [review]

**Pull request merged**